app/helpers/application_helper/button/middleware_server_action.rb

```diff
@@ -1,7 +1,18 @@
 class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
 
   def visible?
-    !@record.present? ||
-      (@record.try(:product) != 'Hawkular' && @record.try(:middleware_server).try(:product) != 'Hawkular')
+    @record.present? && !hawkular_product? && !immutable?
```

Comment:
I think this class would be better written as:

```ruby
class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
  def NullableRecord
    delegate :present?, to: :@record

    def initialize(record)
      @record = record
    end

    def properties
      @record.try(:properties)
    end

    def product
      @record.try(:product).presence ||
        @record.try(:middleware_server).try(:product)
    end
  end


  def visible?
    nullable_record = NullableRecord.new(@record)

    @record.present?
      && !nullable_record.properties['Immutable']
      && nullable_record.product != 'Hawkular'
  end
end
```

That way you gain a more testable interface and make the `visible?` method clearer about its implementation.